src/main/java/org/c4sg/dao/OrganizationDAO.java

```diff
@@ -17,12 +17,45 @@
                                             "WHERE LOWER(o.name) LIKE LOWER(CONCAT('%', :name, '%')) " +
                                                 "OR LOWER(o.description) LIKE LOWER(CONCAT('%', :description, '%')) order by project_updated_time desc";
 
-    String FIND_BY_CRITERIA = "SELECT DISTINCT o FROM Project p RIGHT OUTER JOIN p.organization o " +
+    String FIND_BY_KEYWORD = "SELECT DISTINCT o FROM Project p RIGHT OUTER JOIN p.organization o " +
             "WHERE ((:keyword is null OR LOWER(o.name) LIKE LOWER(CONCAT('%', :keyword, '%')) " +
-                "OR LOWER(o.description) LIKE LOWER(CONCAT('%', :keyword, '%')) OR LOWER(o.country) LIKE LOWER(CONCAT('%', :keyword, '%')))"
-                + " AND (LOWER(o.country) LIKE LOWER(CONCAT('%', :country, '%')) OR :country is null) "
-                + "AND ((LOWER(p.status) ='a' AND :open=true) OR (LOWER(p.status) ='c' AND :open=false) OR :open=false )"
-                + ")  ORDER BY o.name ASC";
+                "OR LOWER(o.description) LIKE LOWER(CONCAT('%', :keyword, '%')) " + 
+                "OR LOWER(o.state) LIKE LOWER(CONCAT('%', :keyword, '%'))" + 
+                "OR LOWER(o.country) LIKE LOWER(CONCAT('%', :keyword, '%'))))" +
+                "AND (:status is null OR o.status = :status) "+
+                "AND (:category is null OR o.category = :category) "+
+                "AND p.id is null "+
+                "  ORDER BY o.name ASC";
+    
+    String FIND_BY_KEYWORD_OPORTUNITES = "SELECT DISTINCT o FROM Project p INNER JOIN p.organization o " +
+            "WHERE ((:keyword is null OR LOWER(o.name) LIKE LOWER(CONCAT('%', :keyword, '%')) " +
+                "OR LOWER(o.description) LIKE LOWER(CONCAT('%', :keyword, '%')) " + 
+                "OR LOWER(o.state) LIKE LOWER(CONCAT('%', :keyword, '%'))" + 
+                "OR LOWER(o.country) LIKE LOWER(CONCAT('%', :keyword, '%'))))" +
+                "AND (:status is null OR o.status = :status) "+
+                "AND (:category is null OR o.category = :category) "+
+                "  ORDER BY o.name ASC";
+    
+    String FIND_BY_KEYWORD_COUNTRIES = "SELECT DISTINCT o FROM Project p RIGHT OUTER JOIN p.organization o " +
+            "WHERE ((:keyword is null OR LOWER(o.name) LIKE LOWER(CONCAT('%', :keyword, '%'))) "+ 
+            "OR LOWER(o.description) LIKE LOWER(CONCAT('%', :keyword, '%'))" + 
+            "OR LOWER(o.state) LIKE LOWER(CONCAT('%', :keyword, '%'))"+
+            "OR LOWER(o.country) LIKE LOWER(CONCAT('%', :keyword, '%')))" + 
+            "AND (o.country in (:country))" +
+            "AND (:status is null OR o.status = :status) "+
+            "AND (:category is null OR o.category = :category) " +
+            "AND p.id is null "+
+            " ORDER BY o.name ASC";
+    
+    String FIND_BY_KEYWORD_COUNTRIES_OPORTUNITES = "SELECT DISTINCT o FROM Project p INNER JOIN p.organization o " +
```

Comment:
Hi Felipe, do you talk about remove static strings and use directly in the query.